### PR TITLE
Mixin models for storing scorecard data nexB#3

### PR DESCRIPTION
**Pull request by 404-geek**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,68 +1,109 @@
...
+    - mit.LICENSE
+    - AUTHORS.rst
+    - README.rst
+    - CONTRIBUTING.rst
+    - CHANGELOG.rst
 
 [options]
-package_dir =
-    =src
+python_requires = >=3.7
```
Comment (AyanSinhaMahapatra) [inline setup.cfg]: Let's make this `3.8` atleast, if not more.

File: setup.cfg [hunk 2]
```diff
@@ -1,68 +1,109 @@
...
+    url
+    package manager
+    package url
 license_files =
-    apache-2.0.LICENSE
-    NOTICE
-    AUTHORS.rst
-    CHANGELOG.rst
-    CODE_OF_CONDUCT.rst
+    - mit.LICENSE
```
Comment (AyanSinhaMahapatra) [inline setup.cfg]: Why use `mit`? We are using `apache-2.0` across aboutcode as standard, this is also in the skeleton.

File: setup.cfg (hunk 1 above)
Comment (AyanSinhaMahapatra) [reply setup.cfg]: I'll update this on the skeleton side too.

File: setup.cfg [hunk 3]
```diff
@@ -1,68 +1,109 @@
...
 install_requires =
 
-
 [options.packages.find]
 where = src
 
+[options.package_data]
+packageurl-python = py.typed
+
+; [options.extras_require]
```
Comment (AyanSinhaMahapatra) [inline setup.cfg:57]: It's fine to use this like in the skeleton under just `testing`, if you think this way is better, we also can change this in the skeleton side for all others. But this should be uniform.

File: setup.cfg [hunk 4]
```diff
@@ -1,68 +1,109 @@
...
+;     black
+;     mypy
+; test =
+;     pytest
+; build =
+;     setuptools
+;     wheel
+; sqlalchemy =
+;     sqlalchemy >= 2.0.0
+
```
Comment (AyanSinhaMahapatra) [inline setup.cfg:69]: We need the documentation parts you removed, we will need to have documentation too through sphinx/rtd

File: src/ossf_scorecard/contrib/models.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+# Visit https://github.com/nexB/ScoreCode for support and
+# download.
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+
+class Package_score_Mixin(models.Model):
```
Comment (AyanSinhaMahapatra) [inline src/ossf_scorecard/contrib/models.py]: We do class names in camelcase, no underscores, and each word first characted capitalized.

File: src/ossf_scorecard/contrib/models.py [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    )
+
+    details = models.JSONField(
+        default=list,
+        blank=True,
+        help_text=_(
+            "A list of details/errors reharding the score"
+        ),
+    )
+
```
Comment (AyanSinhaMahapatra) [inline src/ossf_scorecard/contrib/models.py]: See https://github.com/nexB/ScoreCode/issues/3 we need two things:

1. django mixin models which are not used/populated here
2. normal python classes (maybe with attrs) which store the data here at `src/ossf_scorecard/scorecard.py` and returns this. This has to be used and tested.

File: tests/test_scorecard.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+# tests/test_scorecard.py
+import unittest
+from src.ossf_scorecard.scorecard import get_scorecard
+
+class TestScorecard(unittest.TestCase):
```
Comment (AyanSinhaMahapatra) [inline tests/test_scorecard.py:5]: Let's do the tests on a couple more diverse repos so we make sure we have variety of data and our models work fine in all cases.

**Review by AyanSinhaMahapatra — CHANGES_REQUESTED** [review]
@404-geek Thanks! See comments for changes requested.

File: src/ossf_scorecard/contrib/models.py (hunk 6 above)
Comment (404-geek) [reply src/ossf_scorecard/contrib/models.py]: Hi @AyanSinhaMahapatra I could not understand this part.

Python classes to store the data in the model; should we keep that code in scorecode.

Comment (AyanSinhaMahapatra) [reply src/ossf_scorecard/contrib/models.py]: > Python classes to store the data in the model; should we keep that code in scorecode.

Yes

File: setup.cfg (hunk 1 above)
Comment (404-geek) [reply setup.cfg]: This is updated https://github.com/nexB/ScoreCode/blob/defe444700aed9b659ab024aa9d7bac9c4772f24/setup.cfg#L38

File: setup.cfg (hunk 3 above)
Comment (404-geek) [reply setup.cfg:57]: This is also resolved

File: src/ossf_scorecard/contrib/models.py (hunk 5 above)
Comment (404-geek) [reply src/ossf_scorecard/contrib/models.py]: This is resolved 

https://github.com/nexB/ScoreCode/pull/5/files#diff-56f1d3090dbf7f6d6bf0d756e8ecdbdfe4697289119d8e98e3ce99121e197041R107

File: src/ossf_scorecard/contrib/models.py (hunk 6 above)
Comment (404-geek) [reply src/ossf_scorecard/contrib/models.py]: Please refer to the below link for the object models

https://github.com/nexB/ScoreCode/pull/5/files#diff-56f1d3090dbf7f6d6bf0d756e8ecdbdfe4697289119d8e98e3ce99121e197041

**Pull request merged**